### PR TITLE
Bot enhancements - request updates and release added

### PR DESCRIPTION
**Pull request by ebm5025**
Bot enhancements
    
    - Updated request command to log the requester in the public sheet
    - Added release command to remove pilot and optionally update location
    - Added public sheet key to config

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (sgoodrow) [inline .vscode/launch.json:1]: Nice

File: .gitpod.yml [hunk 2]
```diff
@@ -48,6 +48,7 @@ tasks:
       castleDkpAuctionRaidId: "3431"
       castleDkpBonusesCharId: "3413"
       sharedCharactersGoogleSheetId: "1s-vxZkI2ZK1LAB40MhYI4Vn7oY0YsEiX2vMmetn-SPo"
+      publicCharactersGoogleSheetId: "1hS01upyJZW5_n8ffPCRjro1IS4Z9YwIl7vcSFM9ms7M"
```
Comment (sgoodrow) [inline .gitpod.yml:51]: We should probably rename the above one "private" or something. Another pr tho.

File: src/services/accounts.ts [hunk 3]
```diff
@@ -76,15 +76,15 @@ const authorize = async (sheet: GoogleSpreadsheet) => {
   checkGoogleCredentials();
   return sheet.useServiceAccountAuth({
     client_email: GOOGLE_CLIENT_EMAIL,
-    private_key: (GOOGLE_PRIVATE_KEY || "").replace(/\\n/g, "\n"),
+    private_key:(GOOGLE_PRIVATE_KEY || "").split(String.raw`\n`).join('\n'),
```
Comment (sgoodrow) [inline src/services/accounts.ts:79]: More readable, nice

File: src/services/accounts.ts [hunk 4]
```diff
@@ -76,15 +76,15 @@ const authorize = async (sheet: GoogleSpreadsheet) => {
...
   });
 };
 
 const getAccounts = async () => {
   cache.purgeStale();
   if (cache.size) {
     return cache;
-  }
+  }1
```
Comment (sgoodrow) [inline src/services/accounts.ts:87]: Delete this hehe

File: src/services/public-accounts.ts [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  private async updatePublicAccountSheet(
+    botName: string,
+    cell: SPREADSHEET_COLUMNS,
+    value: any
+  ) {
+    await this.loadBots();
+    if (this.sheet) {
+      const rows = await this.sheet.sheetsByIndex[0].getRows();
+      const botRowIndex = rows.findIndex(
+        (r) => r[SPREADSHEET_COLUMNS.Name] === botName
```
Comment (sgoodrow) [inline src/services/public-accounts.ts:83]: I'm not sure that these two columns across the two spreadsheets align. 🤔

**Review by sgoodrow — APPROVED** [review]
Looks good but please remove the package lock json file since it's redundant w.r.t the yarn lock file.

I'm a little concerned that the mapping between spreadsheets will fail for some bots because the names don't match. I think we'll see the error message if that happens tho and we can fix as needed.

**Comment by ebm5025** [comment]
> Looks good but please remove the package lock json file since it's redundant w.r.t the yarn lock file.
> 
> I'm a little concerned that the mapping between spreadsheets will fail for some bots because the names don't match. I think we'll see the error message if that happens tho and we can fix as needed.

Sounds good, cleaning it up now. I think they definitely need to match up since it should be the same list of names :) But that can be cleaned up if we run into issues.

**Pull request merged**